File: Data Structure/stack/stack.py
# Implement Stack using list
s = [] 

# Insert last
s.append('1')
s.append('2')
s.append('3')
s.append('4')
# print(s)

# Remove last
s.pop() 
s.pop() 
s.pop()
# print(s)

# Implement Stack using collection.deque
from collections import deque

class Stack:
    def __init__(self):
        self.container = deque()
    
    def push(self,val):
        self.container.append(val)
        
    def pop(self):
        return self.container.pop()
    
    def peek(self):
        return  self.container[-1]
    
    def is_empty(self):
        return len(self.container)==0
    
    def size(self):
        return len(self.container)

s = Stack()
s.push(5)
s.pop()
s.is_empty()
s.push(9)
s.push(34)
s.push(78)
s.push(12)

File: Data Structure/stack/stack.js
"use strict";

class Stack {
  constructor() {
    this.items = [];
  }

  // add an item to the stack
  push(item) {
    this.items.push(item);
  }

  // remove and return the removed item
  pop(item) {
    if (this.items.length > 0) {
      return this.items.pop();
    }
  }

  // view the last item
  peek() {
    return this.items[this.items.length - 1];
  }

  isEmpty() {
    return this.items.length === 0;
  }

  size() {
    return this.items.length;
  }

  clear() {
    this.items = [];
  }

  print() {
    console.log(this.items);
    console.log(`The item at the top of the stack is: ${this.peek()}`);
    console.log(
      `Our stack is ${
        this.isEmpty() ? "" : "not"
      } empty; It has length ${this.size()}`
    );
  }
}

/*------- TESTING -----------*/
let stack = new Stack();
stack.push("h");
stack.push("a");
stack.push("p");
stack.push("p");
stack.push("y");
stack.print();

console.log("--------");
stack.pop();
stack.pop();
stack.print();

console.log("--------");
stack.clear();
stack.print();
